app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/id115/ID115Support.java

```diff
@@ -0,0 +1,479 @@
+import nodomain.freeyourgadget.gadgetbridge.model.CallSpec;
+import nodomain.freeyourgadget.gadgetbridge.model.CannedMessagesSpec;
+import nodomain.freeyourgadget.gadgetbridge.model.MusicSpec;
+import nodomain.freeyourgadget.gadgetbridge.model.MusicStateSpec;
+import nodomain.freeyourgadget.gadgetbridge.model.NotificationSpec;
+import nodomain.freeyourgadget.gadgetbridge.model.NotificationType;
+import nodomain.freeyourgadget.gadgetbridge.model.WeatherSpec;
+import nodomain.freeyourgadget.gadgetbridge.service.btle.AbstractBTLEDeviceSupport;
+import nodomain.freeyourgadget.gadgetbridge.service.btle.GattService;
+import nodomain.freeyourgadget.gadgetbridge.service.btle.TransactionBuilder;
+import nodomain.freeyourgadget.gadgetbridge.service.btle.actions.SetDeviceStateAction;
+
+public class ID115Support extends AbstractBTLEDeviceSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(ID115Support.class);
+
+    public BluetoothGattCharacteristic normalWriteCharacteristic = null;
+    public BluetoothGattCharacteristic normalNotifyCharacteristic = null;
+    public BluetoothGattCharacteristic healthWriteCharacteristic = null;
+
+    byte[] currentNotificationBuffer;
+    int currentNotificationSize;
+    int currentNotificationIndex;
+    byte currentNotificationType;
+
+    public ID115Support() {
+        super(LOG);
+        addSupportedService(GattService.UUID_SERVICE_GENERIC_ACCESS);
+        addSupportedService(GattService.UUID_SERVICE_GENERIC_ATTRIBUTE);
+        addSupportedService(ID115Constants.UUID_SERVICE_ID115);
+    }
+
+    @Override
+    protected TransactionBuilder initializeDevice(TransactionBuilder builder) {
+        normalWriteCharacteristic = getCharacteristic(ID115Constants.UUID_CHARACTERISTIC_WRITE_NORMAL);
+        normalNotifyCharacteristic = getCharacteristic(ID115Constants.UUID_CHARACTERISTIC_NOTIFY_NORMAL);
+        healthWriteCharacteristic = getCharacteristic(ID115Constants.UUID_CHARACTERISTIC_WRITE_HEALTH);
+
+        builder.add(new SetDeviceStateAction(getDevice(), GBDevice.State.INITIALIZING, getContext()));
+
+        builder.setGattCallback(this);
+        builder.notify(normalNotifyCharacteristic, true);
+
+        setTime(builder)
+                .setScreenOrientation(builder)
+                .setGoal(builder)
+                .setInitialized(builder);
+
+        return builder;
+    }
+
+    @Override
+    public boolean useAutoConnect() {
+        return true;
+    }
+
+    @Override
+    public void onNotification(NotificationSpec notificationSpec) {
+        sendMessageNotification(notificationSpec);
+    }
+
+    @Override
+    public void onDeleteNotification(int id) {
+
+    }
+
+    @Override
+    public void onSetTime() {
+        try {
+            TransactionBuilder builder = performInitialized("time");
+            setTime(builder);
+            performConnected(builder.getTransaction());
+        } catch(IOException e) {
+        }
```

Comment:
Please don't swallow exception, log them at least.

Reply:
I will fix that.